gridmet_split_script.py

```diff
@@ -0,0 +1,219 @@
+            print('polygon is path to shapefile')
+            polygon_as_bbox = gpd.read_file(polygon_as_bbox)
+            print(polygon_as_bbox.total_bounds)
+        else:
+             raise TypeError('polygon_as_bbox should str path to shapefile or geodataframe')
+
+        if polygon_as_bbox.crs != 'EPSG:4326':
+            polygon_as_bbox = polygon_as_bbox.to_crs('EPSG:4326')
+            print('geodataframe crs changed to EPSG:4326')
+
+        lon_min, lat_min, lon_max, lat_max = polygon_as_bbox.total_bounds
+    else:
+        pass
+
+    ## if only 1 var (short) is provided, change to list
+    if not isinstance(variable, list):
+        variable = [variable]
+
+    ## Initiate list of empty datasets
+    xarray_dict = dict()
+
+    ## Loop through variables and pull data + place in dict
+    for var in variable:
+        ## Pulling data
+        # source: http://thredds.northwestknowledge.net:8080/thredds/reacch_climate_MET_aggregated_catalog.html
+        url = f'http://thredds.northwestknowledge.net:8080/thredds/dodsC/agg_met_{var}_1979_CurrentYear_CONUS.nc'
+        # call data from url
+        start = time.perf_counter()
+        ds = xr.open_dataset(url + '#fillmismatch')
+
+        ## Subset to timeframe and bbox of interest:
+        ds_subset = ds.sel(day = slice(start_date, end_date),
+                           lon = slice(lon_min, lon_max),
+                           lat = slice(lat_max, lat_min))
+        end = time.perf_counter()
+        print(f'finish in {round(start - end, 2)} seconds')
+
+        # Append to dict of xr.datasets
+        xarray_dict[var] = ds_subset
+
+    return xarray_dict
+
+
+def create_weightmap(xarray_dict, polygon, output_data_folder, weightmap_var = None):
+
+    """
+    :param geodataframe polygon: geodataframe polygon or multipolygon
+    :param dict xarray_dict: dictionary of gridmet data. the output of get_gridmet_datasets()
+    :param str output_data_folder: path to folder for function output
+    :param str weightmap_var: variable used to make weightfile + naming of output e.i. weightmap_var = 'tmin'. If none, first variable of dict keys will be used.
+    :return: output path to weightmap pickle file
+    """
+    if isinstance(polygon, geopandas.geodataframe.GeoDataFrame):
+        print('polygon is geodataframe')
+        pass
+    elif os.path.isfile(polygon):
+        print('polygon is path to shapefile')
+        polygon = gpd.read_file(polygon_as_bbox)
+    else:
+        raise TypeError('polygon should str path to shapefile or geodataframe')
+
+    ## Name output weightmap_file
+    if weightmap_var is None:
+        # If non var given
+        weightmap_file = os.path.join(output_data_folder, 'grd2shp_weights.pickle')
+        weightmap_var = list(xarray_dict.keys())[0]
+        print(weightmap_file, weightmap_var)
+    else:
+        weightmap_file = os.path.join(output_data_folder, f'grd2shp_weights_{weightmap_var}.pickle')
+        print(weightmap_file)
```

Comment:
I still don't think you need to have any `weightmap_var` since the weight map is the same for every variable:
```suggestion
    ## Name output weightmap_file
     weightmap_file = os.path.join(output_data_folder, 'grd2shp_weights.pickle')
```

Then you can also take that out of the function definition and the docstring

Reply:
If you do want to keep this, then it might make more sense to 
```suggestion
    ## Name output weightmap_file
    if weightmap_var is None:
         weightmap_var = list(xarray_dict.keys())[0]
     weightmap_file = os.path.join(output_data_folder, f'grd2shp_weights_{weightmap_var}.pickle')
    print(weightmap_file)
```

Reply:
Oh nvm. Maybe it's better to leave as is.